src/py/flwr/server/superlink/state/utils.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utility functions for State."""
+
+
+import time
+from uuid import uuid4
+
+from flwr.common.constant import ErrorCode
+from flwr.proto.error_pb2 import Error  # pylint: disable=E0611
+from flwr.proto.node_pb2 import Node  # pylint: disable=E0611
+from flwr.proto.task_pb2 import Task, TaskIns, TaskRes  # pylint: disable=E0611
+
+NODE_UNAVAILABLE_ERROR_REASON = (
+    "Error: Node Unavailable - The destination node is currently unavailable."
+)
+
+
+def make_node_unavailable_taskres(ref_taskins: TaskIns) -> TaskRes:
+    """Create a TaskRes containing the node available error based on the reference
+    TaskIns."""
+    return TaskRes(
+        task_id=str(uuid4()),
+        group_id=ref_taskins.group_id,
+        run_id=ref_taskins.run_id,
+        task=Task(
+            producer=Node(node_id=0, anonymous=True),
```

Comment:
Open question: should we set the `Node` to anonymous here, or use the `node_id` of the Node that became unavailable?

Reply:
I also had that question. I am inclined to set the `Node` to anonymous here, because the de facto "producer" of this Task is the Superlink. If we set the field to the unavailable node, it would break the convention that the producer_node_id is the place where the task is created. But I might understand the `producer_node_id` wrongly.

Wdyt? @danieljanes @jafermarq